segmentstore/server/src/main/java/io/pravega/segmentstore/server/reading/StorageReadManager.java

```diff
@@ -201,7 +201,12 @@ private Request findOverlappingRequest(Request request) {
     private CompletableFuture<SegmentHandle> getHandle() {
         synchronized (this.lock) {
             if (this.handle == null) {
-                this.handle = storage.openRead(this.segmentName);
+                return storage.openRead(this.segmentName)
```

Comment:
If you get multiple concurrent requests for reading from this segment, this will `openRead` for each such request until one of them succeeds and assigns the handle. 

The previous approach was storing the future in the `this.handle` for the first request coming in, and any requests that come in while `openRead` is executing will piggyback onto that one.

Please fix this code to what it was before, but add an exception handler to reset `this.handle` to `null` if an exception happens during `openRead`.

Reply:
Thanks. Good catch. Totally forgot about that.